### PR TITLE
Delete Related Prices for a deleted ticket and other improvements

### PR DESCRIPTION
**Pull request by manzoorwanijk**
Apart from the below mentioned issue, this PR fixes and improves the price and ticket mutations.
Completes #1990 

## Checklist

* [ ] I have read the documentation relating to systems affected by this pull request, see https://github.com/eventespresso/event-espresso-core/tree/master/docs
* [ ] User input is adequately validated and sanitized
* [ ] all publicly displayed strings are internationalized (usually using `esc_html__()`, see https://codex.wordpress.org/I18n_for_WordPress_Developers)
* [ ] My code is tested.
* [ ] My code follows the Event Espresso code style.
* [ ] My code has proper inline documentation.
* [ ] My code accounts for when the site is in Maintenance Mode (MM2 especially disallows usage of models)

### CODE REVIEW

File: assets/prototype/domain/eventEditor/containers/queries/priceTypes.js [hunk 1]
```diff
@@ -4,13 +4,13 @@ export const PRICE_TYPE_ATTRIBUTES = gql`
...
 		id
 		dbId
+		name
 		baseType
 		isBasePrice
 		isDeleted
 		isDiscount
 		isPercent
 		isTax
-		name
```
Comment (tn3rb) [inline assets/prototype/domain/eventEditor/containers/queries/priceTypes.js]: bah you stinker! I was trying to put them in alphabetical order (except for the IDs) 😆 
but that's ok... not a big deal

File: assets/prototype/domain/eventEditor/ticketsList/ticketCard/useOnDeleteTicket.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+	return () => {
+		const subscriptions = {
+			onCompleted: () => {
+				// The prices that are not default or tax prices.
+				const pricesToDelete = relatedPrices.filter(({ isDefault, isTax }) => !isDefault && !isTax);
+				const priceIdsToDelete = pricesToDelete.map(({ id }) => id);
+				priceIdsToDelete.forEach((id) => {
+					deletePrice({ id });
+				});
```
Comment (tn3rb) [inline assets/prototype/domain/eventEditor/ticketsList/ticketCard/useOnDeleteTicket.js]: could have done the deletions right within the map no?

ie:

```js
pricesToDelete.map(({ id }) => deletePrice({ id }));
```

or were you planning to use those IDs for something else?

File: assets/prototype/domain/eventEditor/ticketsList/ticketPriceCalculator/hooks/useOnSubmitPrices.js [hunk 3]
```diff
@@ -14,41 +14,79 @@ const PRICE_INPUT_FIELDS = [
...
+				}
+				const { id, ...input } = pick(PRICE_INPUT_FIELDS, price);
+				// if it's a newly added price
+				if (!id) {
+					return new Promise((resolve, onError) => {
+						const onCompleted = ({ createPrice: { price } }) => {
+							createdPriceIds.push(price.id);
+							resolve(price);
+						};
+						createEntity({ ...input, ticketId: ticket.id }, { onCompleted, onError });
```
Comment (tn3rb) [inline assets/prototype/domain/eventEditor/ticketsList/ticketPriceCalculator/hooks/useOnSubmitPrices.js]: looks like the field normalization I added got lost 😢

File: assets/prototype/domain/eventEditor/containers/queries/priceTypes.js (hunk 1 above)
Comment (nicolad) [reply assets/prototype/domain/eventEditor/containers/queries/priceTypes.js]: it shouldn't make any difference, 
I also think that we MUST keep them in alphabetical order

Comment (nicolad) [reply assets/prototype/domain/eventEditor/containers/queries/priceTypes.js]: I usually don't have strong opinion in shallow situations like this, but IMO keeping things in alphabetical order is critical to code hygiene

Comment (manzoorwanijk) [reply assets/prototype/domain/eventEditor/containers/queries/priceTypes.js]: Yes, putting them in alphabetical order is better 🤦‍♂

File: assets/prototype/domain/eventEditor/ticketsList/ticketCard/useOnDeleteTicket.js (hunk 2 above)
Comment (manzoorwanijk) [reply assets/prototype/domain/eventEditor/ticketsList/ticketCard/useOnDeleteTicket.js]: Yeah, my idea was to keep track of the mutations which might fail, so we can identify them with those IDs.

File: assets/prototype/domain/eventEditor/ticketsList/ticketPriceCalculator/hooks/useOnSubmitPrices.js (hunk 3 above)
Comment (manzoorwanijk) [reply assets/prototype/domain/eventEditor/ticketsList/ticketPriceCalculator/hooks/useOnSubmitPrices.js]: 🤦‍♂

**Pull request merged**